model-optimizer/extensions/front/tf/TFSliceToSlice.py

```diff
@@ -19,63 +19,48 @@
 from extensions.ops.Cast import Cast
 from extensions.ops.elementwise import Add, Equal
 from extensions.ops.select import Select
+from mo.front.common.partial_infer.utils import int64_array
 from mo.front.common.replacement import FrontReplacementOp
 from mo.graph.graph import Graph, rename_nodes
 from mo.ops.const import Const
+from mo.ops.shape import Shape
 from mo.ops.slice import Slice
+from mo.front.tf.graph_utils import create_op_with_const_inputs
 
 
 class TFSliceToSliceReplacer(FrontReplacementOp):
     """
     This transformation converts TFSlice to internal Slice operation.
-    In TFSlice size[i] == -1 means take all elements on axis i up to the end including(!) the last
-    In internal MO Slice (which is borrowed from ONNX) -1 means take all excluding(!) the last (shape[i] - 1).
-    Also TFSlice has 'sizes' on the second input while Slice has 'ends'.
-    This transformation was added to avoid multiple if statements in future transformations.
+    TFSlice has 'size' on the second input while Slice has 'ends' -> need Add(begin, size)
+    size[i] == -1 is a magic number that means take the whole range along axis i up to the end
+    therefore need to insert subgraph with ShapeOf to process that case
     """
     op = 'TFSlice'
     enabled = True
 
     def replace_sub_graph(self, graph: Graph, match: dict):
-        node = match['op']
-        slice_name = node.soft_get('name', node.id)
+        tf_slice_node = match['op']
+        slice_name = tf_slice_node.soft_get('name', tf_slice_node.id)
         slice_node = Slice(graph).create_node()
-        rename_nodes([(node, slice_name + '/to_be_removed'), (slice_node, slice_name)])
+        rename_nodes([(tf_slice_node, slice_name + '/to_be_removed'), (slice_node, slice_name)])
+        add_node = Add(graph, {'name': slice_name + '/end_const'}).create_node()
 
-        eq_node = Equal(graph, {'name': slice_name + '/equal'}).create_node()
-        minus_one_node = Const(graph, {'name': slice_name + '/minus_one', 'value': np.array(-1)}).create_node()
-        int32_max_node = Const(graph, {'name': slice_name + '/int32_max', 'value': np.iinfo(np.int32).max}).create_node()
-        select_node = Select(graph, {'name': slice_name + '/select'}).create_node()
+        # reconnect begin and input from TFSlice to Slice
+        tf_slice_node.in_port(0).get_connection().set_destination(slice_node.in_port(0))
+        tf_slice_node.in_port(1).get_connection().set_destination(slice_node.in_port(1))
+        tf_slice_node.in_port(2).get_connection().set_destination(add_node.in_port(0))
+        slice_node.in_port(1).get_connection().add_destination(add_node.in_port(1))
 
-        # node to convert sizes to ends
-        sum_node = Add(graph, {'name': slice_name + '/end_const'}).create_node()
+        shapeof_node = Shape(graph, {'name': slice_name + '/ShapeOf'}).create_node()
```

Comment:
I propose to call this node as `max_ends`

Reply:
Renamed to `max_ends`